pkg/domain/infra/abi/manifest.go

```diff
@@ -90,18 +91,34 @@ func (ir *ImageEngine) ManifestInspect(ctx context.Context, name string) ([]byte
 			continue
 		}
 
-		if !manifest.MIMETypeIsMultiImage(manifestType) {
-			appendErr(errors.Errorf("manifest is of type %s (not a list type)", manifestType))
-			continue
-		}
 		result = manifestBytes
 		manType = manifestType
 		break
 	}
 	if len(result) == 0 && latestErr != nil {
 		return nil, latestErr
 	}
-	if manType != manifest.DockerV2ListMediaType {
+
+	switch manType {
+	case manifest.DockerV2Schema2MediaType:
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Printf(`The manifest type %s is not a manifest list. Do you want to inspect the manifest of the single image? [y/N] `, manType)
+		answer, err := reader.ReadString('\n')
+		if err != nil {
+			fmt.Println(errors.Wrapf(err, "error reading input"))
+			os.Exit(1)
+		}
+		if strings.ToLower(answer)[0] != 'y' {
+			return nil, errors.Errorf("manifest is of type %s (not a list type)", manType)
+		}
+		schema2Manifest, err := manifest.Schema2FromManifest(result)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error parsing manifest blob %q as a %q", string(result), manType)
+		}
+		if result, err = schema2Manifest.Serialize(); err != nil {
+			return nil, err
+		}
+	default:
```

Comment:
I really don't like the yes/no answer here, it will be a headache for scripting purposes.  If we don't have a manifest list, can't we just print out a warning and then continue on as if they answered 'Y' here?  "

```
fmt.Printf(`The manifest type %s is not a manifest list.  Displaying the the manifest of the single image.\n`, manType)
```

Reply:
I'm not sure which implementation is more user-friendly. If it's using for scripting porpose, users may expect to receive the manifest list, not a single manifest.

Reply:
Change to just printout the warning.